rust/types/src/metadata.rs

```diff
@@ -97,42 +102,50 @@ MetadataValue
 ===========================================
 */
 
-#[derive(Clone, Debug, PartialEq)]
+#[derive(Clone, Debug, PartialEq, PartialOrd)]
 pub enum MetadataValue {
-    Int(i32),
+    Bool(bool),
+    Int(i64),
     Float(f64),
     Str(String),
-    Bool(bool),
 }
 
-impl TryFrom<&MetadataValue> for i32 {
+impl Eq for MetadataValue {}
```

Comment:
note to self - check if default eq impl is what makes this work

Reply:
Note for context: From my understanding the `Eq` trait is only a marker trait without any implementation. I need this because I hope to use `MetadataValue` as key of a `BTreeMap`, which requires `Ord` and `Eq`.

Reply:
Sorry what I meant to write is that float is not PartialEq, and wanted to reason about that